File: SOLID/Open.py
from abc import ABC, abstractmethod

class Pago(ABC):
    @abstractmethod
    def procesar_pago(self):
        pass

class PagoConTarjeta(Pago):
    def procesar_pago(self):
        print("Procesando pago con tarjeta.")

class PagoConBitcoin(Pago):
    def procesar_pago(self):
        print("Procesando pago con Bitcoin.")

if __name__ == "__main__":
    pago = PagoConTarjeta()
    pago.procesar_pago()

    pago = PagoConBitcoin()
    pago.procesar_pago()

File: SOLID/Open.java

abstract class Pago {
    public abstract void procesarPago();
}

class PagoConTarjeta extends Pago {
    public void procesarPago() {
        System.out.println("Procesando pago con tarjeta.");
    }
}ja

class PagoConBitcoin extends Pago {
    public void procesarPago() {
        System.out.println("Procesando pago con Bitcoin.");
    }
}

public class Open {
    public static void main(String[] args) {
        Pago pago = new PagoConTarjeta();
        pago.procesarPago();

        pago = new PagoConBitcoin();
        pago.procesarPago();
    }
}
